include/irr/asset/ICPUMeshBuffer.h

```diff
@@ -182,24 +183,32 @@ class ICPUMeshBuffer : public IMeshBuffer<ICPUBuffer>, public BlobSerializable,
     @returns Pointer to corresponding buffer's data incremented by `baseVertex` and by `bufferOffset`
     @see @ref getBaseVertex() setBaseVertex() getAttribute()
     */
-    virtual uint8_t* getAttribPointer(const E_VERTEX_ATTRIBUTE_ID& attrId)
+    virtual uint8_t* getAttribPointer(uint32_t attrId)
     {
-        if (!meshLayout)
+        if (!m_pipeline)
+            return nullptr;
+
+        const auto& vtxInputParams = m_pipeline->getVertexInputParams();
+        if (!isAttributeEnabled(attrId))
             return nullptr;
 
-        ICPUBuffer* mappedAttrBuf = static_cast<ICPUMeshDataFormatDesc*>(meshLayout.get())->getMappedBuffer(attrId);
-        if (attrId >= EVAI_COUNT || !mappedAttrBuf)
+        const uint32_t bindingNum = vtxInputParams.attributes[attrId].binding;
+        if (!isVertexAttribBufferBindingEnabled(bindingNum))
+            return nullptr;
+
+        ICPUBuffer* mappedAttrBuf = m_vertexBuffers[bindingNum].get();
+        if (!mappedAttrBuf)
             return nullptr;
 
         int64_t ix = baseVertex;
-        ix *= meshLayout->getMappedBufferStride(attrId);
-        ix += meshLayout->getMappedBufferOffset(attrId);
+        ix *= vtxInputParams.bindings[bindingNum].stride;
+        ix += vtxInputParams.attributes[attrId].offset;
```

Comment:
remember there is also a per-meshbuffer buffer binding offset to be implemented
https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/vkCmdBindVertexBuffers.html

Reply:
resolved